### ISSUE TITLE
panic in unix2dos if the input contains non-ASCII characters followed by newlines

### ISSUE THREAD

**Issue by jrb0001** [spitfire05/rnc#14]
```rust
fn main() {
    newline_converter::unix2dos("ä\n");
}
```

```text
thread 'main' panicked at 'byte index 1 is not a char boundary; it is inside 'ä' (bytes 0..2) of `ä
`', /rustc/897e37553bba8b42751c67658967889d11ecd120\library\core\src\str\mod.rs:678:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\std\src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src/panicking.rs:142:14
   2: core::str::slice_error_fail_rt
   3: core::str::slice_error_fail
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library\core\src\str/mod.rs:86:9
   4: core::str::<impl str>::split_at
             at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\core\src\str/mod.rs:678:13
   5: newline_converter::unix2dos
             at C:\Users\jrb0001\.cargo\registry\src\github.com-1ecc6299db9ec823\newline-converter-0.2.1\src\lib.rs:96:33
   6: inquire_bug::main
             at .\src\main.rs:2:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120\library\core\src\ops/function.rs:248:5
```

**Comment by jrb0001** [comment]
The cause is most likely that the function counts characters and uses that to index into a `str` which is indexed in bytes.

**Comment by spitfire05** [comment]
Good catch, looks like I never tested for anything outside standard ASCII set.

I'll look into fixing that.

**Comment by spitfire05** [comment]
@jrb0001 this should be fixed in 0.2.2.

The bugged versions will be yanked.

**Comment by jrb0001** [comment]
Thanks!
